### PR TITLE
Allow to use ${record} as interpolation source

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Allow to use ${record} in addition to ${tag_parts} and ${hostname} as
interpolated values for sourcetype, source and index.

Switches from the mixin to the fluent provided expander similar to
https://github.com/kazegusuri/fluent-plugin-prometheus/blob/348c112d/lib/fluent/plugin/prometheus.rb

Added various test to ensure the old values and the new values work.

### CODE REVIEW

**Comment by johanneswuerbach** [comment]
@brycied00d friendly ping :-)

File: lib/fluent/plugin/out_splunk-http-eventcollector.rb [hunk 1]
```diff
@@ -25,45 +25,72 @@
...
+          # for v0.14
+          return Fluent::Plugin::RecordTransformerFilter::PlaceholderExpander.new(log: log)
+        else
+          # for v0.12
+          return Fluent::RecordTransformerFilter::PlaceholderExpander.new(log: log)
+        end
+      rescue LoadError => e
+        raise ConfigError, "cannot find filter_record_transformer plugin: #{e.message}"
+      end
+    else # for v0.10, use PlaceholderExapander in fluent-plugin-record-reformer plugin
```
Comment (brycied00d) [inline lib/fluent/plugin/out_splunk-http-eventcollector.rb]: There's really no point in supporting 0.10 here as the gem itself only specifies 0.12, and would only ever conceivably be updated for 0.14.
```ruby
gem.add_runtime_dependency "fluentd", '~> 0.12'
```

**Comment by brycied00d** [comment]
Sorry this sat so long, been busy.

**Comment by johanneswuerbach** [comment]
No worries, thanks for putting this together in the first place :clap:

**Pull request merged**